File: mac0214-course/codeforces/894/b.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int t;
    cin >> t;

    while (t--) {

        int n; cin >> n;
        int b[n];
        for (int i = 0 ; i < n; i++) {
            cin >> b[i]; 
        }

        vector<int> a = {b[0]};

        for (int i = 1; i < n; i++) {
            int x = min(b[i], b[i-1]);
            while (x > b[i] | x >= b[i-1])
                x -= 1;

            if (x >= 1) 
                a.push_back(x);

            a.push_back(b[i]);
        }

        int last = b[n-1];
        if (last > 1)
            a.push_back(last-1);


        int m = a.size();
        cout << m << endl;
        
        for (int i = 0; i < m; i++) {
            cout << a[i];
            if (i != m-1)
                cout << " ";
        } 
        cout << endl;
    }

    return 0;
}

File: mac0214-course/codeforces/894/b.py
t = int(input())

while t > 0:

    n = int(input())

    b  = list(map(int, input().split()))

    a = [b[0]]
    m = 1

    for i in range(1, n):
        x = min(b[i], b[i-1])
        while x > b[i] or x >= b[i-1]:
            x -= 1
        
        if x >= 1:
            a.append(x)
            m += 1
        
        a.append(b[i])
        m += 1
    
    last = b[-1]
    if last > 1:
        a.append(last-1)
        m += 1

    print(m)
    numbers = ""
    for idx, number in enumerate(a):
        numbers += str(number)
        if idx == m-1:
            break
        else:
            numbers += " "

    print(numbers)

    t -= 1